### PR TITLE
samples: caf_sensor_manager: Use sensor stub

### PR DESCRIPTION
**Pull request by emob-nordic**
Clean up the sample, use sensor stub, remove sensor sim support and reorganize file structure for multicore sample.

Signed-off-by: Emil Obalski <Emil.Obalski@nordicsemi.no>

### CODE REVIEW

**Comment by emob-nordic** [comment]
Pushed with release notes updated.

**Comment by emob-nordic** [comment]
@PerMac The doc build failed because i removed label used in release-notes-2.1. Why is it building the doc with that file instead of `release-notes-changelog`? Shouldn't it be this way? Who to ask?

File: samples/caf_sensor_manager/boards/nrf52840dk_nrf52840.overlay [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+ / {
+	sensor_stub: sensor_stub {
+		compatible = "nordic,sensor-stub";
+		label = "SENSOR_STUB";
```
Comment (gmarull) [inline samples/caf_sensor_manager/boards/nrf52840dk_nrf52840.overlay:10]: labels are deprecated, please remove that.

File: samples/caf_sensor_manager/boards/nrf52840dk_nrf52840.overlay [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+		generator = "sensor_stub_gen";
+		status = "okay";
+	};
+
+	agg0: agg0 {
+		compatible = "caf,aggregator";
+		sensor_descr = "accel_sim_xyz";
+		buf_data_length = <240>;
+		sensor_data_size = <24>;
```
Comment (gmarull) [inline samples/caf_sensor_manager/boards/nrf52840dk_nrf52840.overlay:19]: note: this is not following prop naming conventions, should be e.g. `sensor-data-size` (no underscores)

Comment (emob-nordic) [reply samples/caf_sensor_manager/boards/nrf52840dk_nrf52840.overlay:19]: I will fix this in other PR. We didn't know about this convention when we were adding compatible for that.

**Review by gmarull — COMMENTED** [review]
You need to update 2.1.0 release notes,

```
/home/runner/work/sdk-nrf/sdk-nrf/ncs/nrf/doc/_build/nrf/src/releases/release-notes-2.1.0.rst:625:undefined label: 'sensor_stub_config'
```

this has been removed, I guess.

File: samples/caf_sensor_manager/configuration/sensor_manager_def.h [hunk 3]
```diff
@@ -32,24 +32,12 @@ static const struct sm_sampled_channel accel_chan[] = {
...
-	},
-#endif /* IS_ENABLED(CONFIG_SENSOR_SIM) */
-#if IS_ENABLED(CONFIG_SENSOR_STUB)
 	{
 		.dev = DEVICE_DT_GET(DT_NODELABEL(sensor_stub)),
 		.event_descr = DT_PROP(DT_NODELABEL(agg0), sensor_descr),
 		.chans = accel_chan,
 		.chan_cnt = ARRAY_SIZE(accel_chan),
-		.sampling_period_ms = 20,
+		.sampling_period_ms = 200,
```
Comment (nvlsianpu) [inline samples/caf_sensor_manager/configuration/sensor_manager_def.h:40]: Why the sampling period is changing?

Comment (nvlsianpu) [reply samples/caf_sensor_manager/configuration/sensor_manager_def.h:40]: discussed off-line.

**Review by nvlsianpu — APPROVED** [review]

File: samples/caf_sensor_manager/README.rst [hunk 4]
```diff
@@ -8,7 +8,7 @@ CAF: Sensor manager
    :depth: 2
 
 The sensor manager sample demonstrates the functionality of the :ref:`caf_sensor_manager`.
-It uses the :ref:`sensor_sim`, the sensor manager module, the :ref:`caf_sensor_data_aggregator`, and the workload simulator module to automatically sample and gather data and then receive and process it in packages.
+It uses the :ref:`sensor_stub`, the sensor manager module, the :ref:`caf_sensor_data_aggregator`, and the workload simulator module to automatically sample and gather data and then receive and process it in packages.
```
[the remark this answers is not in the document]
Comment (emob-nordic) [reply samples/caf_sensor_manager/README.rst:11]: All except workload module. Maybe we should remove this line completely?

**Review by FrancescoSer — CHANGES_REQUESTED** [review]
Approving but please implement this suggestion

File: doc/nrf/releases/release-notes-changelog.rst [hunk 5]
```diff
@@ -410,6 +410,12 @@ Other samples
     * On the nRF5340 development kit, the :ref:`nrf5340_remote_shell` is now a mandatory sample that must be programmed to the application core.
     * On the nRF5340 development kit, this sample uses the :ref:`shell_ipc_readme` library to forward shell data through the physical application core UART interface.
 
+
+* :ref:`caf_sensor_manager_sample` sample:
+
+  * Removed the sensor sim configuration.
+    The sample now ussed the sensor stub driver by default.
```
Comment (FrancescoSer) [inline doc/nrf/releases/release-notes-changelog.rst:417]:
```suggestion
    The sample now uses the sensor stub driver by default.
```

**Review by FrancescoSer — APPROVED** [review]
Approving but please implement the suggestion

File: doc/nrf/releases/release-notes-changelog.rst (hunk 5 above)
Comment (emob-nordic) [reply doc/nrf/releases/release-notes-changelog.rst]: whopsie. Done

**Pull request merged**